server/src/main/java/alcatraz/server/state/LobbyManager.java

```diff
@@ -1,34 +1,78 @@
 package alcatraz.server.state;
 
+import java.rmi.RemoteException;
 import java.util.HashMap;
+import java.util.UUID;
 
 import alcatraz.shared.Lobby;
+import alcatraz.shared.LobbyKey;
 import alcatraz.shared.Player;
 
 public class LobbyManager {
     private HashMap<Long, Lobby> lobbies;
+    private HashMap<String, Long> lobbyByPlayer;    //MM20241122: Each player can only be registered in one lobby.
+    private long lobbyIdCounter;
+    private static final long MAXSIZE = 100;
 
     public LobbyManager() {
+        this.lobbyIdCounter = 0;
         this.lobbies = new HashMap<>();
+        this.lobbyByPlayer = new HashMap<>();
     }
 
-    public void createLobby(long lobbyId,Lobby lobby) {
-        lobbies.put(lobbyId, lobby);
+    /*
+    * @brief: Construct a new lobby and assign it to the creator's name.
+    *           Several lobbies can be created by the same owner but only the creator will get the secret.
+    *           When an owner creates a new lobby and they have been inside a lobby before, they will be relocated to
+    *               the newly created lobby.
+     */
+    public LobbyKey createLobby(Player owner) throws RemoteException {
+        if (lobbies.size() == MAXSIZE) {
+            throw new RemoteException();        //MM20241122: find better Exception! "MAXSIZE reached"
+        }
+
+        while (lobbies.containsKey(lobbyIdCounter)) {
```

Comment:
Is it really more efficient to look for a first free id, instead of generating random lobby id? 🤔

Reply:
it was the simplest mechanism, that I could think of and that also works